crates/hotshot/src/traits/networking/combined_network.rs

```diff
@@ -381,6 +456,31 @@ impl<TYPES: NodeType> ConnectedNetwork<Message<TYPES>, TYPES::SignatureKey>
         <Libp2pNetwork<_, _> as ConnectedNetwork<Message<TYPES>,TYPES::SignatureKey>>::
             inject_consensus_info(self.secondary(), event).await;
     }
+
+    async fn update_view(&self, view: &u64) {
+        let mut map_lock = self.delayed_tasks.write().await;
+        while let Some((first_view, _task)) = map_lock.first_key_value() {
+            if first_view < view {
+                if let Some((_view, task)) = map_lock.pop_first() {
+                    cancel_task(task).await;
```

Comment:
I think we do not want to do other async actions while holding the write lock.  It can lead to a deadlock.  I think it's ok here.  It's probably safer to aggregate the tasks to cancel and then cancel them.